File: 452. Minimum Number of Arrows to Burst Balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a,b) -> {
            if (a[1] > b[1]) return 1;
            else return -1;
        });
        int count = 0;
        int i = 0;
        while (i < points.length) {
            int end = points[i][1];
            count++;
            while (i < points.length && points[i][0] <= end) i++;
        }
        return count;
    }
}

File: 452. Minimum Number of Arrows to Burst Balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[0])
        count = 0
        for i in range(len(points)):
            if i == 0:
                count += 1
                end = points[i][1]
            else:
                if points[i][0] <= end:
                    end = min(end, points[i][1])
                else:
                    count += 1
                    end = points[i][1]
        return count